src/withEditorProps/index.js

```diff
@@ -42,19 +90,51 @@ export const handleSave = props => (opts = {}) => {
   const updateLastSavedIdToCurrent = () => {
     lastSavedIdUpdate(sequenceData.stateTrackingId);
   };
-  const promiseOrVal =
-    (!readOnly || alwaysAllowSave || opts.isSaveAs) &&
-    saveHandler &&
-    saveHandler(
-      opts,
-      tidyUpSequenceData(sequenceData, { annotationsAsObjects: true }),
-      props,
-      updateLastSavedIdToCurrent
-    );
 
-  if (promiseOrVal && promiseOrVal.then) {
-    return promiseOrVal.then(updateLastSavedIdToCurrent);
-  }
+  const save = () => {
+    // TODO: pass additionalProps (blob or error) to the user
+    const promiseOrVal =
+      (!readOnly || alwaysAllowSave || opts.isSaveAs) &&
+      saveHandler &&
+      saveHandler(
+        opts,
+        tidyUpSequenceData(sequenceData, { annotationsAsObjects: true }),
+        props,
+        updateLastSavedIdToCurrent
+      );
+
+    if (promiseOrVal && promiseOrVal.then) {
+      return promiseOrVal.then(updateLastSavedIdToCurrent);
+    }
+  };
+
+  // We don't want to make save slower for users that don't want
+  // a png as output
+  // if (!generatePng) {
```

Comment:
This will be the option to pass to the Editor component to generate a png onsave.